### DISCUSSION TITLE
PageFind Sorting

### DISCUSSION THREAD

**Discussion by Karlstens** [ATQQ/sugar-blog#245] [Q&A] [answered]
I am looking for guidance to implement PageFind sorting. Although the search returns, they don't appear to be sorted;

```
Example sort of the shape <span data-pagefind-sort="date:2024-01-01">rectangle</span>.
```

```
Another example sort of the shape <span data-pagefind-sort="date:2022-01-01">rectangle</span>.
```

I'm not sure on how to implement the next step;
https://pagefind.app/docs/js-api-sorting/

```
const search = await pagefind.search("static", {
    sort: {
        date: "asc"
    }
});
```

The term "rectangle" returns, but doesn't appear as per the desired sort order.

**Comment by ATQQ** [comment]
When the plugin is built, it only injects some necessary `pagefind` properties into the pages.

What kind of sorting criteria do you roughly need? I am planning to provide a `sort` method similar to the current `filter` method. Do you think it can meet your sorting needs?

**Reply by Karlstens** [reply]
The sort that I had in mind is by using either an ISO date (such as GIT commit time), or an application version string found within each page FrontMatter, to be cast against popular search terms - so that terms with a more recent date/newer version are found earlier in search returns. 

Page 1 returns first due to the more recent date.
```
Example sort of the shape <span data-pagefind-sort="date:2024-01-01">rectangle</span>.
```

Page 2 returns after due to being older;
```
Another example sort of the shape <span data-pagefind-sort="date:2022-01-01">rectangle</span>.
```

I did try to implement this below - but couldn't get it working via VitePress config;

docs\.vitepress\config.mjs
```
import { defineConfig } from "vitepress";
import { pagefindPlugin } from "vitepress-plugin-pagefind";

export default defineConfig({
	title: "My VitePress Project",
	description: "Testing VitePress",
	lastUpdated: true,

	vite: {
		plugins: [pagefindPlugin({
				searchOptions: {
					sort: {
						date: "desc", // Change to "desc" if you want descending order
					},
				},
			}),],
	},

	themeConfig: {
	...
	}
	...
})
```

My understanding (_and you may correct me if I'm mistaken_) is that sort is very similar to weighted results? I'm assuming both weighted results (_which are working last time I tested_) could also work in parallel with a sort method - however, I'm not too sure which would be prioritized first, either the weight and then sort, or the sort and then weight. 

At this point in time, I've only been reading through the concepts posted on the official PageFind website.

**Reply by ATQQ** [reply]
Understood, I plan this sort to be a method, which is array.sort, where array is the pagefind search results.

The data of each item is the same as the current filter, mainly the value on the frontmatter. Of course, I will also add the date, which is also the time obtained through Git. Now getting the date is an option. You can see that there is a `showDate` option. , turning it on will take extra time to build.

**Comment by ATQQ** [comment, marked as answer]
now vitepress-plugin-pagefind@0.4.2 support sort option

```js
pagefindPlugin({
  sort(a, b) {
    return b.meta.date - a.meta.date
  },
})
```

you can also set `showDate: true` show date suffix
```js
pagefindPlugin({
  showDate: true
})
```
<img width="587" alt="image" src="https://github.com/ATQQ/sugar-blog/assets/42485491/8051db44-3707-478f-b269-37e167d9c7c2">
